### DISCUSSION TITLE
Schema capturing in maxwell

### DISCUSSION THREAD

**Discussion by Shukla-Ankur** [zendesk/maxwell#2056] [Q&A]
This thread is to understand how maxwell builds schemas.
Fom the documentation, I understand that 3 tables capture the "base" schema and then newer changes are captured in schemas table. 
Assume I start maxwell on MySQL server for 1st time, and I am using init_position to start playing binlogs from an old time T1. There are DDLs after time T1 also. 
1.  In this case, what is the "base" schema captured in database,table and column tables captured? Is it the current schema of tables in the databases? 
2. If base schema captured is current schema, then how does maxwell manage to process events from an older point in time where schema would be different for some tables?

**Comment by osheroff** [comment]
Hi Shukla, I'm gonna move this thread to discussions.

**Reply by osheroff** [reply]
1. Yes, the current schema is what's captured as the base schema.  There's no way to "go back" in time.

2. It doesn't.  You're really only guaranteed to be able to process events from the initial point that you capture the schema.  Maxwell's bootstrapping feature is designed to fill in the gaps there.

**Comment by osheroff** [comment]
now technically this isn't *100%* the way it has to be: with `binlog_row_metadata=FULL`, I *think* maxwell could honestly operate totally without even having to capture the schema.  I haven't coded support for that yet tho.
